frontend/src/routes/(authenticated)/org/[org_id]/BulkAddOrgMembers.svelte

```diff
@@ -0,0 +1,159 @@
+  import { OrgRole, type BulkAddOrgMembersResult } from '$lib/gql/types';
+  import t from '$lib/i18n';
+  import { z } from 'zod';
+  import { _bulkAddOrgMembers } from './+page';
+  import { AdminContent } from '$lib/layout';
+  import Icon from '$lib/icons/Icon.svelte';
+  import BadgeList from '$lib/components/Badges/BadgeList.svelte';
+  import { distinct } from '$lib/util/array';
+  import { SupHelp, helpLinks } from '$lib/components/help';
+  import { usernameRe } from '$lib/user';
+  import OrgMemberBadge from '$lib/components/Badges/OrgMemberBadge.svelte';
+  import type { UUID } from 'crypto';
+
+  enum BulkAddSteps {
+    Add,
+    Results,
+  }
+
+  let currentStep = BulkAddSteps.Add;
+
+  export let orgId: string;
+  const schema = z.object({
+    usernamesText: z.string().trim().min(1, $t('org_page.bulk_add_members.empty_user_field')),
+  });
+
+  let formModal: FormModal<typeof schema>;
+  $: form = formModal?.form();
+
+  let addedMembers: BulkAddOrgMembersResult['addedMembers'] = [];
+  let notFoundMembers: BulkAddOrgMembersResult['notFoundMembers'] = [];
+  let existingMembers: BulkAddOrgMembersResult['existingMembers'] = [];
+
+  function validateBulkAddInput(usernames: string[]): FormSubmitReturn<typeof schema> {
+    if (usernames.length === 0) return { usernamesText: [$t('org_page.bulk_add_members.empty_user_field')] };
+
+    for (const username of usernames) {
+      if (username.includes('@')) {
+        if (!isEmail(username)) return { usernamesText: [$t('org_page.bulk_add_members.invalid_email_address', { email: username })] };
+      } else if (!usernameRe.test(username)) {
+        return { usernamesText: [$t('org_page.bulk_add_members.invalid_username', { username })] };
+      }
+    }
+  }
+
+  async function openModal(): Promise<void> {
+    currentStep = BulkAddSteps.Add;
+    const { response } = await formModal.open(undefined, async (state) => {
+      const usernames = state.usernamesText.currentValue
+        .split('\n')
+        // Remove whitespace
+        .map(s => s.trim())
+        // Remove empty lines before validating, otherwise final newline would count as invalid because empty string
+        .filter(s => s)
+        .filter(distinct);
+
+      const bulkErrors = validateBulkAddInput(usernames);
+      if (bulkErrors) return bulkErrors;
+
+      const { error, data } = await _bulkAddOrgMembers(
+        orgId as UUID,
+        usernames,
+        OrgRole.User,
+      );
+
+      addedMembers = data?.bulkAddOrgMembers.bulkAddOrgMembersResult?.addedMembers ?? [];
+      notFoundMembers = data?.bulkAddOrgMembers.bulkAddOrgMembersResult?.notFoundMembers ?? [];
+      existingMembers = data?.bulkAddOrgMembers.bulkAddOrgMembersResult?.existingMembers ?? [];
+      return error?.message;
+    }, { keepOpenOnSubmit: true });
+
+    if (response === DialogResponse.Submit) {
+      currentStep = BulkAddSteps.Results;
+    }
+  }
+</script>
+
+<AdminContent>
```

Comment:
we want this to be visible to org admins too, so remove this admin content component

Reply:
Done in commit 2980f90a.